Blish HUD/Controls/Textbox.cs

```diff
@@ -1,233 +1,136 @@
+        /// </summary>
         public event EventHandler<EventArgs> EnterPressed;
-        public event EventHandler<Microsoft.Xna.Framework.Input.Keys> KeyPressed;
-        public event EventHandler<Microsoft.Xna.Framework.Input.Keys> KeyDown;
-        public event EventHandler<Microsoft.Xna.Framework.Input.Keys> KeyUp;
 
-        protected System.Windows.Forms.TextBox _mttb;
+        protected virtual void OnEnterPressed(EventArgs e) => EnterPressed?.Invoke(this, e);
 
-        public string Text {
-            get => _mttb.Text;
-            set => _mttb.Text = value;
-        }
-
-        protected string _placeholderText;
-        public string PlaceholderText {
-            get => _placeholderText;
-            set => SetProperty(ref _placeholderText, value);
-        }
-
-        protected Color _foreColor = Color.FromNonPremultiplied(239, 240, 239, 255);
-        public Color ForeColor {
-            get => _foreColor;
-            set => SetProperty(ref _foreColor, value);
-        }
+        private int _prevCursorIndex  = 0;
+        private int _horizontalOffset = 0;
 
-        private TimeSpan _lastInvalidate;
-        private bool _textWasChanged = false;
+        public TextBox() {
+            _multiline = false;
+            _maxLength = 2048;
 
-        protected bool _caretVisible = false;
-        private bool CaretVisible {
-            get => _caretVisible;
-            set => SetProperty(ref _caretVisible, value);
+            this.Size = new Point(STANDARD_CONTROLWIDTH, STANDARD_CONTROLHEIGHT);
         }
 
-        protected BitmapFont _font = Content.DefaultFont14;
-        public BitmapFont Font {
-            get => _font;
-            set => SetProperty(ref _font, value);
+        protected override void HandleEnter() {
+            OnEnterPressed(EventArgs.Empty);
         }
 
-        /// <summary>
-        /// [NOT THREAD-SAFE]
-        /// </summary>
-        public TextBox() {
-            _lastInvalidate = DateTime.MinValue.TimeOfDay;
-
-            _mttb = new System.Windows.Forms.TextBox() {
-                Parent                   = BlishHud.Form,
-                Size                     = new Size(20, 20),
-                Location                 = new System.Drawing.Point(-500),
-                AutoCompleteMode         = AutoCompleteMode.Append,
-                AutoCompleteSource       = AutoCompleteSource.CustomSource,
-                AutoCompleteCustomSource = new AutoCompleteStringCollection(),
-                ShortcutsEnabled         = true,
-                TabStop                  = false
-            };
-
-            _sharedInterceptor.LeftMouseButtonReleased += delegate {
-                if (_sharedInterceptor.ActiveControl == this) {
-                    Textbox_LeftMouseButtonReleased(null, null);
-                }
-            };
+        protected override void MoveLine(int delta) { /* NOOP */ }
```

Comment:
Shouldn't this set the cursor position at the beginning and at the end?

Reply:
Fixed by checking the delta and calling either `HandleHome(false)` or `HandleEnd(false)`.